linalg/hypre.cpp

```diff
@@ -1587,6 +1640,41 @@ void HypreParMatrix::Destroy()
    }
 }
 
+/* job =  0, extract block diagonal of A and scale A into C
+ * job =  1, job 0 + scale b into d
+ * job =  2, use A to scale b only
+ */
+#if MFEM_HYPRE_VERSION >= 21800
+int BlockInverseScale(const HypreParMatrix *A, HypreParMatrix *C,
+                      const Vector *b, HypreParVector *d, int block, int job)
+{
+   if (0 == job || 1 == job)
+   {
+      hypre_ParCSRMatrix *C_hypre;
+      hypre_ParcsrBdiagInvScal(*A, block, &C_hypre);
+      hypre_ParCSRMatrixDropSmallEntries(C_hypre, 1e-15, 1);
+      (*C).WrapHypreParCSRMatrix(C_hypre);
```

Comment:
```suggestion
      C->WrapHypreParCSRMatrix(C_hypre);
```

Reply:
Re colCSRMat, because it is a CSR matrix for all (global) columns and local rows. Happy to change if you have a better suggestion.